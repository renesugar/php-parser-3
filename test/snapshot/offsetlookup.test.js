const parser = require('../main');

describe("offsetlookup", function() {
  it("simple", function() {
    expect(parser.parseEval('$obj["index"];')).toMatchSnapshot();
  });
  it("variable", function() {
    expect(parser.parseEval('$obj[$var];')).toMatchSnapshot();
  });
  it("call", function() {
    expect(parser.parseEval('$obj[$var]();')).toMatchSnapshot();
  });
  it("multiple", function() {
    expect(parser.parseEval('$obj["first"]["second"];')).toMatchSnapshot();
  });
});
