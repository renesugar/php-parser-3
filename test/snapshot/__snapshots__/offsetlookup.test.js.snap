// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offsetlookup call 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [],
        "kind": "call",
        "what": OffsetLookup {
          "kind": "offsetlookup",
          "offset": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "var",
          },
          "what": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "obj",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`offsetlookup multiple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": OffsetLookup {
        "kind": "offsetlookup",
        "offset": String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"second\\"",
          "unicode": false,
          "value": "second",
        },
        "what": OffsetLookup {
          "kind": "offsetlookup",
          "offset": String {
            "isDoubleQuote": true,
            "kind": "string",
            "raw": "\\"first\\"",
            "unicode": false,
            "value": "first",
          },
          "what": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "obj",
          },
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`offsetlookup simple 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": OffsetLookup {
        "kind": "offsetlookup",
        "offset": String {
          "isDoubleQuote": true,
          "kind": "string",
          "raw": "\\"index\\"",
          "unicode": false,
          "value": "index",
        },
        "what": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "obj",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`offsetlookup variable 1`] = `
Program {
  "children": Array [
    ExpressionStatement {
      "expression": OffsetLookup {
        "kind": "offsetlookup",
        "offset": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "var",
        },
        "what": Variable {
          "byref": false,
          "curly": false,
          "kind": "variable",
          "name": "obj",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
